internal/indexmanagement/reconcile.go

```diff
@@ -362,6 +378,27 @@ func (imr *IndexManagementRequest) reconcileIndexManagementCronjob(policy apis.I
 		log.V(1).Info("Skipping rollover management for policymapping; hot phase not defined", "policymapping", mapping.Name)
 	}
 
+	// prune-namespaces cron job
+	if policy.Phases.Delete != nil && policy.Phases.Delete.Namespaces != nil {
```

Comment:
Checking `len(policy.Phases.Delete.Namespaces) != 0` is more useful here. It ensures that if your slice is nil or empty at the same time, the prune job will not be scheduled.